File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/models/Command.java
package com.inspiredandroid.linuxcommandbibliotheca.models;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Created by simon on 24.01.16.
 */
public class Command extends RealmObject {
    public final static String ID = "id";
    public final static String NAME = "name";
    public final static String DESCRIPTION = "desc";

    @PrimaryKey
    private int id;
    private int category;
    @Required
    private String name;
    @Required
    private String desc;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCategory() {
        return category;
    }

    public void setCategory(int category) {
        this.category = category;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }
}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/models/Command.kt
package com.inspiredandroid.linuxcommandbibliotheca.models

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

/* Copyright 2019 Simon Schubert
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
open class Command : RealmObject() {

    @PrimaryKey
    var id: Int = 0
    var category: Int = 0
    var name: String = ""
    var desc: String = ""

    companion object {
        const val ID = "id"
        const val NAME = "name"
        const val DESCRIPTION = "desc"
    }
}
